server/src/main/java/io/unitycatalog/server/service/AuthService.java

```diff
@@ -156,7 +154,14 @@ public HttpResponse grantToken(OAuthTokenExchangeRequest request) {
             .tokenType(AuthTypes.BEARER)
             .build();
 
-    return HttpResponse.ofJson(response);
+    Cookie cookie = Cookie.secureBuilder("UC_TOKEN", accessToken).path("/").build();
```

Comment:
Also, I was thinking that by default, lets still be more Oauth2 compliant and not set a header, but when we see the query parameter `ext=cookie` then set it. So @jamieknight-db would call `auth/tokens?ext=cooke` to get a response that sets a cookie.

Reply:
Add the condition to set based on the query param